test/dotnet/install_dotnet_lts.sh

```diff
@@ -13,7 +13,7 @@ source dev-container-features-test-lib
 source dotnet_env.sh
 source dotnet_helpers.sh
 
-expected=$(fetch_latest_version_in_channel "LTS")
+expected=$(fetch_latest_version_in_channel "8.0")
```

Comment:
This approach doesn't make sense.

As discussed this offline,

URL used by tests is https://github.com/devcontainers/features/blob/main/test/dotnet/dotnet_helpers.sh#L16C9-L16C87

However, for LTS we use  https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/dotnet-helpers.sh#L57-L61 & https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/dotnet-helpers.sh#L79-L82

We don’t seem to directly utilize any URL for finding LTS by the Feature but depend on the vendor script (which we have no control over).
Check if URLs mentioned in https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/vendor/dotnet-install.sh#L1301-L1302 or https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/vendor/dotnet-install.sh#L1311-L1312 might be making any difference and update the tests if we could sync!

Reply:
both the urls mentioned here:
https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/vendor/dotnet-install.sh#L1301-L1302 or https://github.com/devcontainers/features/blob/main/src/dotnet/scripts/vendor/dotnet-install.sh#L1311-L1312 
are not working to retrieve the latest sdk version of dotnet i.e. 8.0.302

can you suggets what we should update in the test?

Reply:
Hi, I also had a look at this problem and it seems the install-script uses `https://aka.ms/dotnet/$channel`, e.g.

```
https://aka.ms/dotnet/LTS/dotnet-sdk-linux-x64.tar.gz
```

This starts the download of version SDK 8.0.302 while [latest.version](https://dotnetcli.azureedge.net/dotnet/Sdk/LTS/latest.version) from the dotnetcli CDN is stuck on 8.0.301.

I would like to report this issue, but I don't know which github repo is best for this. Maybe [dotnet ](https://github.com/dotnet/core)?

Reply:
Actually they don't really want anyone to use the `latest.version` file (see https://github.com/dotnet/core/issues/9280#issuecomment-2082867900).

The recommended API for retrieving versions is
<https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json>

You can parse it with `jq`, for example:

``` sh
curl -s https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json |
    jq --raw-output '[."releases-index"[] | select(."release-type"=="lts" and ."support-phase"=="active") ."latest-sdk"] | first'```

Reply:
Thanks @@sliekens. I made changes accordingly and now the test is not failing.